src/Mobs/Ghast.h

```diff
@@ -17,10 +17,18 @@ class cGhast :
 
 	CLASS_PROTODEF(cGhast)
 
+	virtual void Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk) override;
 	virtual void GetDrops(cItems & a_Drops, cEntity * a_Killer = nullptr) override;
 	virtual bool Attack(std::chrono::milliseconds a_Dt) override;
 
-	bool IsCharging(void) const {return false; }
+	bool IsCharging(void) const { return m_bIsCharging; }
+
+private:
+	// Specifies whether or not the ghast has started shooting a fireball
+	bool m_bIsCharging;
+	
+	// Used to determine when to shoot a fireball, from the moment the ghast started charging
+	int m_ChargeTimer;
```

Comment:
```suggestion
	/** Number of ticks until a projectile is created.
	Only used while m_IsCharging is true. */
	int m_TicksUntilShot;
```
(combined with inverting the logic - set `m_TicksUntilShot` to 10 in `Attack()` and decrease until 0 in `Tick()` )